pandas/core/arrays/array_.py

```diff
@@ -0,0 +1,113 @@
+import numpy as np
+
+from pandas._libs import lib, tslibs
+
+from pandas.core.dtypes.common import is_extension_array_dtype
+from pandas.core.dtypes.dtypes import registry
+from pandas.core.dtypes.generic import ABCIndexClass, ABCSeries
+
+from pandas import compat
+
+
+def array(data, dtype=None, copy=False):
+    """
+    Create an array.
+
+    .. versionadded:: 0.24.0
+
+    Parameters
+    ----------
+    data : Sequence[object]
+        A sequence of scalar instances for `dtype`. The underlying
+        array will be extracted from a Series or Index object.
+
+    dtype : Union[str, np.dtype, ExtensionDtype], optional
+        The dtype to use for the array. This may be a NumPy
+        dtype, or an extension type registered with pandas using
+        :meth:`pandas.api.extensions.register_extension_dtype`.
+
+        If not specified, there are two possibilities:
+
+        1. When `data` is a :class:`Series`, :class:`Index`, or
```

Comment:
did you add tests for cases where you specify an EA (either thru Series, or directoy) and a dtype is specified? This could be a *different* dtype, what do we do then? .astype? (I think we should), e.g. this is analagous to

```
In [1]: arr = np.array([1,2])

In [2]: np.array(arr, dtype=float)
Out[2]: array([1., 2.])
```

we should mention this in the doc-string and test at least somewhat for this.

Reply:
I added a test.

I'm not sure it's worth mentioning in the docstring. It feels like a corner case, and is covered by the first line of the dtype description: `The dtype to use for the array. `